service_darwin.go

```diff
@@ -184,28 +185,76 @@ func (s *darwinLaunchdService) Uninstall() error {
 }
 
 func (s *darwinLaunchdService) Status() (Status, error) {
-	exitCode, out, err := runWithOutput("launchctl", "list", s.Name)
-	if exitCode == 0 && err != nil {
-		if !strings.Contains(err.Error(), "failed with stderr") {
+	var (
+		// Scan output for a valid PID.
+		re        = regexp.MustCompile(`"PID" = ([0-9]+);`)
+		isRunning = func(out string) bool {
+			matches := re.FindStringSubmatch(out)
+			return len(matches) == 2 // `PID = 1234`
+		}
+		canIgnore = func(err error) bool {
+			return err == nil ||
+				strings.Contains(err.Error(),
+					"Could not find service")
+		}
+	)
+
+	// Get output from `list`.
+	_, out, err := runWithOutput("launchctl", "list", s.Name)
+	if !canIgnore(err) {
+		return StatusUnknown, err
+	}
+	if isRunning(out) {
+		return StatusRunning, nil
+	}
+
+	// root only sees "system daemons"
+	// `list` will not see "user daemons" (which we likely are).
+	// If we're impersonating another user (like root via `sudo` or setuid bit),
+	// try again as the user ID that called us.
+	var (
+		impersonating       bool
+		targetID            string
+		uid                 = syscall.Getuid()
+		suidStr, sudoExeced = os.LookupEnv("SUDO_UID")
+		euid                = syscall.Geteuid()
+	)
+	if sudoExeced {
+		suid, err := strconv.Atoi(suidStr)
+		if err != nil {
 			return StatusUnknown, err
 		}
+		impersonating = impersonating || suid != euid
+		targetID = suidStr
+	} else {
+		impersonating = uid != euid
+		targetID = strconv.Itoa(uid)
 	}
 
-	re := regexp.MustCompile(`"PID" = ([0-9]+);`)
-	matches := re.FindStringSubmatch(out)
-	if len(matches) == 2 {
-		return StatusRunning, nil
+	if impersonating {
+		_, out, err := runWithOutput("sudo", "-u", "#"+targetID,
+			"launchctl", "list", s.Name)
+		if !canIgnore(err) {
+			return StatusUnknown, err
+		}
+		if isRunning(out) {
+			return StatusRunning, nil
+		}
 	}
 
+	// `list` will always return a "job" entry if the job is "loaded"
+	// (see launchd documentation for keyword semantics)
+	// We'll check if the plist actually exist to determine if it's installed or not.
+	// And since it's not running, we can assume it's stopped (but still installed).
 	confPath, err := s.getServiceFilePath()
 	if err != nil {
 		return StatusUnknown, err
 	}
-
-	if _, err = os.Stat(confPath); err == nil {
+	if _, err := os.Stat(confPath); err == nil {
```

Comment:
This error needs to be inspected. As-is it's silently dropped.
If it's "not exist" we're fine, otherwise we need to bail. (Permission issues, etc.)